starlark/interp.go

```diff
@@ -81,6 +81,17 @@ func (fn *Function) CallInternal(thread *Thread, args Tuple, kwargs []Tuple) (Va
 
 	var iterstack []Iterator // stack of active iterators
 
+	// Use defer so that application panics can pass through
+	// interpreter without leaving thread in a bad state.
+	defer func() {
+		// ITERPOP the rest of the iterator stack.
+		for _, iter := range iterstack {
+			iter.Done()
+		}
+
+		fr.locals = nil
```

Comment:
Nit: Do you think it's more readable to move this to a separate defer immediately after fr.locals is initialized?

Reply:
Not really, and more importantly I don't want to add the cost of an additional deferred call to the loop.